paimon-flink/paimon-flink-common/src/test/java/org/apache/paimon/flink/SchemaChangeITCase.java

```diff
@@ -28,7 +32,211 @@
+        assertThat(result.toString())
+                .contains(
+                        "CREATE TABLE `PAIMON`.`default`.`T` (\n"
+                                + "  `a` VARCHAR(2147483647) NOT NULL,\n"
+                                + "  `b` VARCHAR(2147483647),\n"
+                                + "  `c` VARCHAR(2147483647),\n"
+                                + "  `d` INT,");
+
+        sql("ALTER TABLE T DROP (c, d)");
+        result = sql("SHOW CREATE TABLE T");
+        assertThat(result.toString())
+                .contains(
+                        "CREATE TABLE `PAIMON`.`default`.`T` (\n"
+                                + "  `a` VARCHAR(2147483647) NOT NULL,\n"
+                                + "  `b` VARCHAR(2147483647),");
+    }
+
+    @Test
+    public void testRenameColumn() {
+        sql("CREATE TABLE T (a STRING PRIMARY KEY NOT ENFORCED, b STRING, c STRING)");
+        sql("INSERT INTO T VALUES('paimon', 'bbb', 'ccc')");
+        sql("ALTER TABLE T RENAME c TO c1");
+        List<Row> result = sql("SHOW CREATE TABLE T");
+        assertThat(result.toString())
+                .contains(
+                        "CREATE TABLE `PAIMON`.`default`.`T` (\n"
+                                + "  `a` VARCHAR(2147483647) NOT NULL,\n"
+                                + "  `b` VARCHAR(2147483647),\n"
+                                + "  `c1` VARCHAR(2147483647)");
+        result = sql("SELECT a, b, c1 FROM T");
+        assertThat(result.toString()).isEqualTo("[+I[paimon, bbb, ccc]]");
+
+        // column do not exist.
+        assertThatThrownBy(() -> sql("ALTER TABLE T RENAME d TO d1"))
+                .hasMessageContaining("The column `d` does not exist in the base table.");
+
+        // target column exist.
+        assertThatThrownBy(() -> sql("ALTER TABLE T RENAME a TO b"))
+                .hasMessageContaining("The column `b` already existed in table schema.");
+    }
+
+    @Test
+    public void testModifyColumnType() {
+        sql(
+                "CREATE TABLE T (a STRING PRIMARY KEY NOT ENFORCED, b STRING, c STRING, d INT, e FLOAT)");
+        sql("ALTER TABLE T MODIFY e DOUBLE");
+        List<Row> result = sql("SHOW CREATE TABLE T");
+        assertThat(result.toString())
+                .contains(
+                        "CREATE TABLE `PAIMON`.`default`.`T` (\n"
+                                + "  `a` VARCHAR(2147483647) NOT NULL,\n"
+                                + "  `b` VARCHAR(2147483647),\n"
+                                + "  `c` VARCHAR(2147483647),\n"
+                                + "  `d` INT,\n"
+                                + "  `e` DOUBLE,");
+
+        assertThatThrownBy(() -> sql("ALTER TABLE T MODIFY c DOUBLE"))
+                .getRootCause()
+                .isInstanceOf(IllegalStateException.class)
+                .hasMessage(
+                        "Column type c[STRING] cannot be converted to DOUBLE without loosing information.");
```

Comment:
The message `Column type c[STRING] cannot be converted to DOUBLE without loosing information` is from
`SchemaManager` Line 291. I believe there is a typo error. I think it should be `without losing information`.